src/gt4py/gtscript_imports.py

```diff
@@ -192,25 +185,44 @@ def install(
     search_path: Optional[List[Union[str, pathlib.Path]]] = None,
     generate_path: Optional[Union[str, pathlib.Path]] = None,
     in_source: bool = False,
-):
+) -> GtsFinder:
     """
     Install GTScript import extensions.
 
-    Parameters
-    ----------
-    search_path : `List[Pathlike]`
-        A list of paths where gtscript files are expected to be found.
-        Also added to the search path for python files.
-
-    generate_path : `Pathlike`, optional
-        Path to a directory where python modules generated from `gtscript` files
-        will be created. Can be used to cache the process between runs.
-        If none is given, a temporary directory is used.
-
-    in_source: `bool`
-        If true, `generate_path` will be ignored and python modules will be
-        generated alongside the corresponding `gtscript` sources.
+    Parameters are passed through to the constructor of :py:class:`GtsFinder`.
     """
     finder = GtsFinder(search_path=search_path, generate_path=generate_path, in_source=in_source)
     finder.install()
     return finder
+
+
+@contextmanager
+def allow_import_gtscript(**kwargs: Any) -> Iterator:
+    """
+    Create a context within which GTScript extensions can be imported.
+
+    Example
+    -------
+    .. code-block: python
+
+        mystencil = None
+        with allow_import_gtscript(search_path=[pathlib.Path("my/gtscript/extensions/")]):
```

Comment:
Anything that speaks against using a string right away for the path?

If the module would be called differently we could use `gtscript_imports` instead of `allow_import_gtscript` for the context handler.

Reply:
I am thinkgin of renaming to `installed` ->
```python
gtscript_imports.install()
...
# or
with gtscript_imports.installed():
    ...
```